source/includes/code-examples/GridFS.cs

```diff
@@ -0,0 +1,415 @@
+        {
+            // ASCII for "HelloWorld"
+            byte[] bytes = { 72, 101, 108, 108, 111, 87, 111, 114, 108, 100 };
+
+            for (int i = 0; i < 5; i++)
+            {
+                uploader.Write(bytes, 0, bytes.Length);
+            }
+
+            uploader.Close();
+        }
+        // end-open-upload-stream-with-options
+    }
+
+    static async Task UploadStreamAsync()
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+
+        // Uploads data to a stream, then writes the stream to a GridFS file
+        // start-upload-from-stream-async
+        using (var fileStream = new FileStream("/path/to/input_file", FileMode.Open, FileAccess.Read))
+        {
+            await bucket.UploadFromStreamAsync("new_file", fileStream);
+        }
+        // end-upload-from-stream-async
+    }
+
+    static void UploadStream()
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+
+        // Uploads data to a stream, then writes the stream to a GridFS file
+        // start-upload-from-stream
+        using (var fileStream = new FileStream("/path/to/input_file", FileMode.Open, FileAccess.Read))
+        {
+            bucket.UploadFromStream("new_file", fileStream);
+        }
+        // end-upload-from-stream
+    }
+
+    static void Find()
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+
+        // Prints information about each file in the bucket
+        // start-find
+        var filter = Builders<GridFSFileInfo>.Filter.Empty;
+
+        var files = bucket.Find(filter);
+
+        foreach (var file in files.ToEnumerable())
+        {
+            Console.WriteLine(file.ToJson());
+        }
+        // end-find
+    }
+
+    static async Task FindAsync()
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+
+        // Prints information about each file in the bucket
+        // start-find-async
+        var filter = Builders<GridFSFileInfo>.Filter.Empty;
+
+        var files = await bucket.FindAsync(filter);
+
+        await files.ForEachAsync(file => { Console.WriteLine(file.ToJson()); });
```

Comment:
The lambda passed to `ForEachAsync` should actually be async!

Reply:
can you explain what this looks like?

Reply:
An async lambda should return a Task.

The initial problem here is that `Console.WriteLine` is sync and `Console` does not have an async version for `WriteLine`.

You could try something like this:

```await files.ForEachAsync(file => Console.Out.WriteLineAsync(file.ToJson()))```